### PR TITLE
Fix python-sdk package requirements

### LINKED ISSUE #6792: Bug: Python `lakefs_sdk` doesn't automatically install dependencies
When running `pip install lakefs-sdk`, some dependencies are missing.

Steps to reproduce:

1. In an empty directory the following `Dockerfile`:

```dockerfile
FROM python:3.10
RUN apt update -y && apt upgrade -y
RUN pip install lakefs-sdk
CMD ["python", "-c", "import lakefs_sdk"]
```

2. Build it:

```bash
$ docker build -t lakefs_sdk_test .
```

3. Run it:

```bash
$ docker run --rm -it lakefs_sdk_test
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/lakefs_sdk/__init__.py", line 21, in <module>
    from lakefs_sdk.api.actions_api import ActionsApi
  File "/usr/local/lib/python3.10/site-packages/lakefs_sdk/api/__init__.py", line 4, in <module>
    from lakefs_sdk.api.actions_api import ActionsApi
  File "/usr/local/lib/python3.10/site-packages/lakefs_sdk/api/actions_api.py", line 20, in <module>
    from pydantic import validate_arguments, ValidationError
ModuleNotFoundError: No module named 'pydantic'
```

I can also reproduce this locally in a virtualenv on MacOS with Python from Homebrew.

Additionally, the sdk won't work with latest pydantic - it requires a specific range of version to work, so naively running `pip install pydantic` won't help.

### PR DESCRIPTION
**Pull request by nopcoder**
Fix #6792

### CODE REVIEW

File: clients/python/setup.py [hunk 1]
```diff
@@ -21,10 +21,12 @@
...
 # http://pypi.python.org/pypi/setuptools
-
+PYTHON_REQUIRES = ">=3.7"
 REQUIRES = [
-  "urllib3 >= 1.25.3",
-  "python-dateutil",
+    "urllib3 >= 1.25.3, < 2.1.0",
+    "python-dateutil",
+    "pydantic >= 1.10.5, < 2",
+    "aenum"
```
Comment (Jonathan-Rosenberg) [inline clients/python/setup.py:29]: I think that it should go to `clients/python/templates/setup.mustache`

**Review by Jonathan-Rosenberg — CHANGES_REQUESTED** [review]
The comment in the files

**Comment by nopcoder** [comment]
> The comment in the files

Applied the fix to the template

**Review by Jonathan-Rosenberg — APPROVED** [review]
LGTM

**Pull request merged**